### PR TITLE
fraction coordinates on segment and section

### PR DESCRIPTION
**Pull request by eleftherioszisis**
These functions permit the extraction of the 3D coordinates that correspond to a fraction of the length of a segment or along the piecewise linear curve of the section. @dkeller9 asked me to implement these functions for they are useful for synapse location calculations.

### CODE REVIEW

File: neurom/analysis/morphmath.py [hunk 1]
```diff
@@ -47,6 +47,49 @@ def vector(p1, p2):
     return np.subtract(p1[0:3], p2[0:3])
 
 
+def two_points_fraction(p1, p2, fraction):
```
Comment (juanchopanza) [inline neurom/analysis/morphmath.py]: How about "interpolate"?

File: neurom/sections.py [hunk 2]
```diff
@@ -92,6 +92,20 @@ def area(section):
...
+    of the path length along the piecewise linear curve which
+    is constructed from the section points.
+
+    Args:
+        0, 1, 2 correspoding to 3D cartesian coordinates
+
+    Returns:
+        The 3D coordinates of the aforementioned point
+    '''
+    return mm.path_fraction_point(tuple(n.value[:3] for n in section), fraction)
```
Comment (juanchopanza) [inline neurom/sections.py]: Do you really need to slide here? I think you can just pass the points as values.

File: neurom/segments.py [hunk 3]
```diff
@@ -73,6 +73,21 @@ def identity(segment):
...
+    '''Computes the point p between two points p1, p2
+    for which |p1p| = fraction * |p1p2|
+
+    Args:
+        fraction: float between 0. and 1.
+
+    Returns:
+        The 3D coordinates of the aforementioned point
+    '''
+    return mm.two_points_fraction(segment[0].value[:3],
```
Comment (juanchopanza) [inline neurom/segments.py]: Not sure you need to slice here.

File: neurom/analysis/morphmath.py (hunk 1 above)
Comment (eleftherioszisis) [inline neurom/analysis/morphmath.py]: Hmm, I think interpolate too general. Interpolation can be linear, polynomial,etc.

linear_interpolate maybe?

**Comment by lidakanari** [comment]
What about the diameters - radii? Do we need an interpolation there as well?

File: neurom/segments.py (hunk 3 above)
Comment (eleftherioszisis) [inline neurom/segments.py]: You' re right. Lemme see if removing the slicing is ok for section above as well.

File: neurom/analysis/morphmath.py (hunk 1 above)
Comment (juanchopanza) [inline neurom/analysis/morphmath.py]: That's fine.

**Comment by eleftherioszisis** [comment]
This is an interesting point. @dkeller9, would it be useful if the interpolated diameter is returned as well?

Also it needs a different name if it returns the radius as well. Any ideas?

**Comment by eleftherioszisis** [comment]
Ok, I added the segment radius interpolation along with the 3D point. I renamed the segment function ```cross_section_at_fraction```

**Comment by juanchopanza** [comment]
@eleftherioszisis  Looks OK to me. I can merge if you're ready to go.

**Comment by eleftherioszisis** [comment]
I don't have anything else to add. Feel free to merge it!

**Pull request merged**